src/Lean/Linter/Sets.lean

```diff
@@ -0,0 +1,34 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+prelude
+import Lean.Elab.Command
+import Lean.Linter.Basic
+
+namespace Lean.Linter
+
+/-- Add a new linter set that contains the given linters. -/
+def insertLinterSet [MonadEnv m] (setName : Name) (linterNames : Array Name) : m Unit :=
+  modifyEnv (linterSetsExt.addEntry · (setName, linterNames))
+
+/-- `registerSet` wraps `registerOption` by setting relevant values. -/
+def registerSet (setName : Name) (ref : Name := by exact decl_name%) : IO (Lean.Option Bool) := do
+  registerOption setName {
+    declName := ref
+    defValue := false
+    group := "linterSet"
+    descr := ""
+  }
+  return { name := setName, defValue := false }
+
+open Lean.Elab.Command in
+/-- Declare a new linter set by giving the set of options that will be enabled along with the set. -/
+elab doc?:(docComment)? "register_linter_set" name:ident " := " decl:ident* : command => do
```

Comment:
What do you think of a command like `#show_linter_sets` that prints out which are the registered linter sets, maybe with "foldable" entries, so that you see the name of each registered linter set, the value of the option and you can click on it to see which options it contains?

Reply:
(I think this is a great idea, by the way.)

Reply:
It is a great suggestion! It sounds quite complicated though and I'm not familiar how to do fancy trace messages with the unfolding, so I propose to leave this to future work.